src/components/dialogs/create-case-dialog.tsx

```diff
@@ -28,45 +28,74 @@ import UploadNewCase from './commons/upload-new-case';
 import { ElementType } from '../../utils/elementType';
 
 import { useSnackMessage } from '@gridsuite/commons-ui';
-import { NameCheckReturn, useNameCheck } from './commons/useNameCheck';
+import { NameCheckReturn, useNameCheck } from './commons/use-name-check';
+import { FormProvider, useForm } from 'react-hook-form';
+import { CASE_FILE, CASE_NAME, DESCRIPTION } from '../utils/field-constants';
 
 const MAX_FILE_SIZE_IN_MO = 100;
 const MAX_FILE_SIZE_IN_BYTES = MAX_FILE_SIZE_IN_MO * 1024 * 1024;
 
-interface CreateCaseDialogProps {
+interface ICreateCaseDialogProps {
     onClose: () => void;
     open: boolean;
 }
 
-const CreateCaseDialog: React.FC<CreateCaseDialogProps> = ({
+const CreateCaseDialog: React.FunctionComponent<ICreateCaseDialogProps> = ({
     onClose,
     open,
 }) => {
     const intl = useIntl();
     const dispatch = useDispatch();
     const { snackError } = useSnackMessage();
 
+    const createCaseFormMethods = useForm({
+        defaultValues: {
+            [CASE_NAME]: '',
+            [DESCRIPTION]: '',
+            [CASE_FILE]: null,
+        },
```

Comment:
Where is the validation with yup ?

Reply:
J'ai ajouter la validation yup. thank you